smart-contracts/contracts/PublicLock.sol

```diff
@@ -205,6 +216,29 @@ contract PublicLock is ILockCore, ERC165, IERC721, IERC721Receiver, Ownable {
     return _purchaseFor(_recipient, _referrer, _data);
   }
 
+  /**
+   * @dev Destroys the user's key and sends a refund based on the amount of time remaining.
+   */
+  function cancelAndRefund()
+    external
+  {
+    _cancelAndRefund(msg.sender);
+  }
+  
+  /**
+   * @dev Destroys the user's key and sends a refund based on the amount of time remaining.
+   * The caller must first be approved to modify the owner's key.
+   * @param _owner The owner of the key to be destroyed.
+   */
+  function cancelAndRefundFor(
```

Comment:
Is that a realistic use case?
I mean if the user has to approve someone else first, why can't they cancel at the same time?

Reply:
I'm in favour of removing this for now for the sake of simplicity', and only adding it if it becomes necessary down the road.

Reply:
I can remove it.
The thinking was if someone is already approved to trade away my Key, they could choose to cancel instead which is effectively another way of trading the Key for ETH.

Reply:
Ah, indeed, but again, if I am willing to get a refund I can just cancel myself rather than authorize someone else, right?

Reply:
Removed